### PR TITLE
Update analytics binary to be compatible with the testing harness.

### PR DESCRIPTION
**Pull request by f2prateek**
https://paper.dropbox.com/doc/Libraries-End-to-End-Testing-Harness-dnotyIVIGYR7lnO1LcZtM#:uid=457845578223014078216812&h2=Contract

Main changes are:
* writeKey is a flag instead of an env var
* method is a flag, instead of a top level command
* context, properties, traits etc are optional

Also fixed a bug with `screen` in the binary. Previously it was invoking `identify` instead of screen.

### CODE REVIEW

**Comment by rohitpaulk** [comment]
Will take a look at this tonight 👍

File: bin/analytics [hunk 1]
```diff
@@ -10,143 +10,82 @@ program :name, 'simulator.rb'
 program :version, '0.0.1'
 program :description, 'scripting simulator'
 
-# use an env var for write key, instead of a flag
-Analytics = Segment::Analytics.new({
-  write_key: ENV['SEGMENT_WRITE_KEY'],
-  on_error: Proc.new { |status, msg| print msg }
-})
-
 def toObject(str)
```
Comment (rohitpaulk) [inline bin/analytics]: I'd suggest renaming this to `to_hash`. `toObject` looks like it was copied from a JS implementation..

File: bin/analytics [hunk 2]
```diff
@@ -10,143 +10,82 @@ program :name, 'simulator.rb'
...
-      properties: toObject(options.properties),
-      context: toObject(options.context),
-      category: options.category
-       })
-    Analytics.flush
-  end
+# analytics -method=<method> -segment-write-key=<segmentWriteKey> [options]
 
-end
+command :"" do |c|
```
Comment (rohitpaulk) [inline bin/analytics]: Nitpick: The common syntax for this is to do `command :something ... ` followed by `default_command :something`. [ref](https://github.com/commander-rb/commander#command-defaults)

File: bin/analytics [hunk 3]
```diff
@@ -10,143 +10,82 @@ program :name, 'simulator.rb'
...
-      anonymous_id: options.anonymous,
-      traits: toObject(options.traits),
-      context: toObject(options.context)
-       })
-    Analytics.flush
-  end
-
-end
+  c.option '--writeKey=<writeKey>', String, 'the Segment writeKey'
+  c.option '--method=<method>', String, 'the Segment writeKey'
```
Comment (rohitpaulk) [inline bin/analytics]: Wrong description for `method`

Comment (f2prateek) [reply bin/analytics]: Updated!

File: bin/analytics (hunk 2 above)
Comment (f2prateek) [reply bin/analytics]: Updated!

File: bin/analytics (hunk 1 above)
Comment (f2prateek) [reply bin/analytics]: I couldn't get `to_hash` to work for some reason. Possibly related to https://github.com/Shopify/ssllabs.rb/pull/31/files ? Instead I named it `json_hash`, but open to other suggestions.

```
→ go run cmd/cli/main.go -segment-write-key='OnMMoZ6YVozrgSBeZ9FpkC0ixH0ycYZn' -runscope-token='92f6b74c-0200-4735-9195-702d528aa97f' -runscope-bucket='bmbbsjxdttr7' -path='../analytics-ruby/bin/analytics'
../analytics-ruby/bin/analytics:13:in `to_hash': wrong number of arguments (given 0, expected 1) (ArgumentError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:408:in `gsub'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:408:in `regexp'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:412:in `candidate'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:429:in `candidate'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:434:in `complete'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:839:in `complete'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:1701:in `block in visit'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:1700:in `reverse_each'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:1700:in `visit'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:1732:in `block in complete'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:1731:in `catch'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:1731:in `complete'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:1541:in `block in parse_in_order'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:1534:in `catch'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:1534:in `parse_in_order'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:1528:in `order!'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:1620:in `permute!'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/optparse.rb:1642:in `parse!'
	from /Library/Ruby/Gems/2.3.0/gems/commander-4.4.4/lib/commander/runner.rb:381:in `parse_global_options'
	from /Library/Ruby/Gems/2.3.0/gems/commander-4.4.4/lib/commander/runner.rb:65:in `run!'
	from /Library/Ruby/Gems/2.3.0/gems/commander-4.4.4/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.3.0/gems/commander-4.4.4/lib/commander/import.rb:5:in `block in <top (required)>'
main[17633]: 2018-03-06 15:26:50.003 - cli/main.go:38 - test error: could not invoke command: exit status 1
exit status 1

```

**Comment by f2prateek** [comment]
@rohitpaulk merging for now, but let me know if there are any changes you'd like to see!

**Pull request merged**